### PR TITLE
fix `The document.open function throws an error because document.defaultView returns null` (close #1272)

### LINKED ISSUE #1272: The document.open function throws an error because document.defaultView returns null
Error message: `Uncaught TypeError: Cannot read property 'top' of null`
Find on http://www.virgilio.it/.
Example for reproducing:
```html
<body>
<script>
	var counter = 0;
	var getHtml = function () {
		return '<div>Hello ' + (counter++) + '</div>';
	};
	
	var loadHandler = function (iframe) {
		var win = iframe.contentWindow;
		var doc = iframe.contentDocument;
		
		win.__renderer__ || (win.__renderer__ = true, setTimeout(function () {
			doc.open();
			doc.write(getHtml());
			win.__renderer__ = true;
			doc.close();
		}, 0));
	}
	
	var div = document.createElement('div');
	
	div.innerHTML = '<iframe src="javascript:\'\';" onload="eval(\'(\' + loadHandler' +
					'.toString() + \')(this);\')"></iframe>';
	
	var iframe = div.firstChild;
	
	document.body.appendChild(iframe);
</script>
</body>
```

### PR DESCRIPTION
**Pull request by LavrovArtem**
/cc @miherlosev @churkin

### CODE REVIEW

File: test/client/fixtures/sandbox/node/document-test.js [hunk 1]
```diff
@@ -494,22 +495,56 @@ if (!browserUtils.isIE) {
...
-    document.body.appendChild(iframe);
+            var selfRemovingScripts = nativeMethods.querySelectorAll.call(iframeDocument,
+                '.' + SHADOW_UI_CLASSNAME.selfRemovingScript);
+
+            strictEqual(selfRemovingScripts.length, 0);
+        });
+});
 
-    var iframeDocument = iframe.contentDocument;
+test('should not throw an error when document.defualtView is null', function () {
```
Comment (miherlosev) [inline test/client/fixtures/sandbox/node/document-test.js]: `test(should not throw an error when document.defualtView is null (GH-1272))`

File: test/client/fixtures/sandbox/node/document-test.js [hunk 2]
```diff
@@ -494,22 +495,56 @@ if (!browserUtils.isIE) {
...
-    iframeDocument.open();
-    iframeDocument.write('<body style=\"padding: 0; margin: 0; overflow: hidden;\"></body>');
-    iframeDocument.close();
+        iframe.id     = 'test' + Date.now();
+        iframe.src    = 'javascript:"";';
+        iframe.onload = function () {
+            var doc = iframe.contentDocument;
 
-    strictEqual(nativeMethods.querySelectorAll.call(document, '.' + SHADOW_UI_CLASSNAME.selfRemovingScript).length, 0);
+            // NOTE: Without setTimeout the error does not reproduced
```
Comment (miherlosev) [inline test/client/fixtures/sandbox/node/document-test.js]: `// NOTE: Without wrapping in 'setTimeout' function the error is not reproduced`

**Comment by LavrovArtem** [comment]
ping @churkin

**Comment by LavrovArtem** [comment]
ping @churkin

**Comment by LavrovArtem** [comment]
ping @churkin

**Pull request merged**